cmd/skopeo/logout.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"io"
+
+	"github.com/containers/common/pkg/auth"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type logoutOptions struct {
+	global     *globalOptions
+	logoutOpts *auth.LogoutOptions
+}
+
+func logoutCmd(global *globalOptions) *cobra.Command {
+	opts := logoutOptions{
+		global:     global,
+		logoutOpts: &auth.LogoutOptions{},
+	}
+	cmd := &cobra.Command{
+		Use:     "logout",
+		Short:   "Logout of a container registry",
+		Long:    "Logout of a container registry on a specified server.",
+		RunE:    commandAction(opts.run),
+		Example: `skopeo logout quay.io`,
+	}
+	usageHelper(cmd)
+	cmd.Flags().AddFlagSet(auth.GetLogoutFlags(opts.logoutOpts))
+	return cmd
+}
+
+func (opts *logoutOptions) run(args []string, stdout io.Writer) error {
+	if len(args) > 1 {
+		return errors.Errorf("too many arguments, logout takes at most 1 argument")
+	}
+	if len(args) == 0 && !opts.logoutOpts.All {
+		return errors.Errorf("registry must be given")
+	}
+
+	var server string
+	if len(args) == 1 {
+		server = auth.GetRegistryName(args[0])
+	}
+	sys, err := opts.global.newSystemContext()
+	if err != nil {
+		return err
+	}
+	sys.AuthFilePath = opts.logoutOpts.Authfile
```

Comment:
Why is the set of options in `login` and here so different?

Reply:
These options are the current design of podman/buildah logout

Reply:
Oh, right; logout does not do anything over the network, so the TLS-related options are irrelevant.